### PR TITLE
Transaction and Hashed orderings

### PR DESCRIPTION
**Pull request by TheMMaciek**

### CODE REVIEW

File: modules/shared/src/main/scala/org/tessellation/schema/transaction.scala [hunk 1]
```diff
@@ -73,11 +73,21 @@ object transaction {
...
   }
 
+  implicit val transactionOrder: Order[Transaction] = (x: Transaction, y: Transaction) =>
+    implicitly[Order[BigInt]].compare(x.ordinal.coerce, y.ordinal.coerce)
+
+  implicit val transactionOrdering: Ordering[Transaction] = transactionOrder.toOrdering
+
   implicit val signedTransactionOrder: Order[Signed[Transaction]] = (x: Signed[Transaction], y: Signed[Transaction]) =>
-    implicitly[Order[BigInt]].compare(x.value.ordinal.coerce, y.value.ordinal.coerce)
+    implicitly[Order[Transaction]].compare(x.value, y.value)
```
Comment (marcinwadon) [inline modules/shared/src/main/scala/org/tessellation/schema/transaction.scala:82]: hmm, I see there is already a generic order for `Signed[A]`. maybe you could reuse it? 🤔

File: modules/shared/src/main/scala/org/tessellation/schema/transaction.scala [hunk 2]
```diff
@@ -73,11 +73,21 @@ object transaction {
...
 
   implicit val signedTransactionOrdering: Ordering[Signed[Transaction]] = signedTransactionOrder.toOrdering
 
+  implicit val hashedTransactionOrder: Order[Hashed[Transaction]] = (x: Hashed[Transaction], y: Hashed[Transaction]) =>
```
Comment (marcinwadon) [inline modules/shared/src/main/scala/org/tessellation/schema/transaction.scala:86]: the same for `Hashed`. if there is no generic one, you can create one

File: modules/shared/src/main/scala/org/tessellation/security/Hashed.scala [hunk 3]
```diff
@@ -8,3 +10,11 @@ import derevo.derive
 
 @derive(eqv, show)
 case class Hashed[A <: AnyRef](signed: Signed[A], hash: Hash, proofsHash: ProofsHash)
+
+object Hashed {
+  implicit def autoUnwrap[T <: AnyRef](t: Hashed[T]): T = t.signed
```
Comment (TheMMaciek) [inline modules/shared/src/main/scala/org/tessellation/security/Hashed.scala:15]: FYI it uses `autoUnwrap` from `Signed` also to give the `T`. Not sure if it should use `Signed.autoUnwrap` or unwrap `Signed` here on its own.

Comment (marcinwadon) [reply modules/shared/src/main/scala/org/tessellation/security/Hashed.scala:15]: imo it's fine

**Review by marcinwadon — APPROVED** [review]

**Pull request merged**